prog/ai/inference_endpoint_nexus.rb

```diff
@@ -51,7 +51,13 @@ def self.assemble(project_id:, location:, boot_image:, name:, vm_size:, storage_
       firewall = internal_project.firewalls_dataset.where(location: location).where(Sequel[:firewall][:name] => "inference-endpoint-firewall").first
       fail "No firewall named 'inference-endpoint-firewall' configured for inference endpoints in #{location}" unless firewall
       subnet_s = Prog::Vnet::SubnetNexus.assemble(internal_project.id, name: ubid.to_s, location: location, firewall_id: firewall.id)
-      lb_s = Prog::Vnet::LoadBalancerNexus.assemble(subnet_s.id, name: ubid.to_s, src_port: 443, dst_port: 8443, health_check_endpoint: "/up", health_check_protocol: "https")
+
+      custom_dns_zone = DnsZone.where(project_id: Config.inference_endpoint_service_project_id).where(name: "ai.ubicloud.com").first
+      custom_hostname_prefix = if custom_dns_zone
+        name + (is_public ? "" : "-#{ubid.to_s[-5...]}")
```

Comment:
nit: if it's private, maybe we can just use regular lb hostname?

Reply:
there's this problem: the name of the load balancer currently is the ubid of the inference endpoint. So the regular lb hostname is ugly, something like `1eprwnxxm9wt7gyhbm9az8sq1y.ycx1b.lb.ubicloud.com`. We cannot use `name.ycx1b.lb.ubicloud.com` because all load balancers for inference endpoints are in an internal project and we mandate that names in a project are unique. That means that a customer cannot use `name` if another customer is using that name already, even though these two load balancers would get different hostnames as they each are in a different subnet.